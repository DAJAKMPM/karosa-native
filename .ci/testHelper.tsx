import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
