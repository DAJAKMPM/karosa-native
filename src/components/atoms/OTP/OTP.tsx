/**
 *
 * OTP
 * @format
 *
 */

import React, { FC, useState } from "react";
import { TextInput, View } from "react-native";

import type { PropsType } from "./types";
import OTPStyles from "./styles";

const OTP: FC<PropsType> = (props) => {
  const { inputLength, value, onChangeText } = props;

  const [currentValue, setValue] = useState<string>(value || "");

  return (
    <View style={OTPStyles.container}>
      {new Array(inputLength).fill(0).map((_, index) => {
        return (
          <TextInput
            editable={false}
            selectTextOnFocus={false}
            style={[
              OTPStyles.txtOTP,
              // eslint-disable-next-line react-native/no-inline-styles
              { opacity: currentValue?.charAt(index) ? 1 : 0.5 },
            ]}
            value={currentValue?.charAt(index) || ""}
            key={index}
          />
        );
      })}

      <TextInput
        maxLength={inputLength}
        keyboardType="number-pad"
        returnKeyType="done"
        onChangeText={(text) => {
          setValue(text);
          if (onChangeText) onChangeText(text);
        }}
        style={OTPStyles.hiddenText}
      />
    </View>
  );
};

export default OTP;
