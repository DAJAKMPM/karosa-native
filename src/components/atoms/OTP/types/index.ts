/**
 *
 * @format
 *
 */
import { TextInputProps } from "react-native";

export type PropsType = {
  inputLength: number;
} & TextInputProps;
