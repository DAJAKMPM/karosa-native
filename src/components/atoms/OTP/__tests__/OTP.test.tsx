import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import OTP from "../OTP";

describe("OTP Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<OTP inputLength={2} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
