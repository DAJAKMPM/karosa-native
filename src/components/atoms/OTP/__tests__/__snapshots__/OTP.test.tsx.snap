// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTP Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 50,
        "maxHeight": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      rejectResponderTermination={true}
      selectTextOnFocus={false}
      style={
        Array [
          Object {
            "borderBottomColor": "#000",
            "borderBottomWidth": 2,
            "color": "#0C0C0C",
            "fontFamily": "SFProText-Bold",
            "fontSize": 24,
            "fontWeight": "400",
            "lineHeight": 28,
            "marginHorizontal": 5,
            "textAlign": "center",
            "width": 46,
          },
          Object {
            "opacity": 0.5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      editable={false}
      rejectResponderTermination={true}
      selectTextOnFocus={false}
      style={
        Array [
          Object {
            "borderBottomColor": "#000",
            "borderBottomWidth": 2,
            "color": "#0C0C0C",
            "fontFamily": "SFProText-Bold",
            "fontSize": 24,
            "fontWeight": "400",
            "lineHeight": 28,
            "marginHorizontal": 5,
            "textAlign": "center",
            "width": 46,
          },
          Object {
            "opacity": 0.5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      maxLength={2}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "height": "100%",
          "opacity": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
